var/spack/repos/builtin/packages/abacus/package.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import re
+
+from spack import *
+
+
+class Abacus(MakefilePackage):
+    """ABACUS (Atomic-orbital Based Ab-initio Computation at UStc)
+    is an open-source computer code package aiming
+    for large-scale electronic-structure simulations
+    from first principles"""
+
+    maintainers = ["bitllion"]
+
+    homepage = "http://abacus.ustc.edu.cn/"
+    git = "https://github.com/abacusmodeling/abacus-develop.git"
+    url = "https://github.com/abacusmodeling/abacus-develop/archive/refs/tags/v2.2.1.tar.gz"
+
+    version("develop", branch="develop")
+    version(
+        "2.2.1",
+        sha256="14feca1d8d1ce025d3f263b85ebfbebc1a1efff704b6490e95b07603c55c1d63",
+    )
+    version(
+        "2.2.0",
+        sha256="09d4a2508d903121d29813a85791eeb3a905acbe1c5664b8a88903f8eda64b8f",
+    )
+
+    variant("mpi", default=True, description="Builds with MPI support. Requires MPI2+")
+    variant(
+        "openmp", default=True, description="Enables OpenMP threads. Use threaded FFTW3"
+    )
+
+    depends_on("elpa+openmp")
+    depends_on("cereal")
+    depends_on("libxc")
+    depends_on("fftw")
+    depends_on("mpi", when="+mpi", type=("build", "link", "run"))
```

Comment:
Given the comment for the variant, I *think* what you want is something along the lines:

```suggestion
    depends_on("mpi@2:", when="+mpi", type=("build", "link", "run"))
```

Reply:
Sorry, I don't quite understand the meaning of spack's grammar. Thank you for your suggestions.
Abacus's MPI dependency is necessary, so I deleted the MPI parameters "when" and "variant".
The variable OpenMP is related to the dependent elpa. I modified the logical judgment of elpa when using OpenMP.